client/llb/fileop.go

```diff
@@ -54,7 +56,7 @@ type CopyInput interface {
 }
 
 type subAction interface {
-	toProtoAction(context.Context, string, pb.InputIndex) (pb.IsFileAction, error)
+	toProtoAction(ctx context.Context, parent string, base pb.InputIndex, platform string) (pb.IsFileAction, error)
```

Comment:
This is `os`, not `platform`.

Do we need this in LLB client at all? Paths are normalized before marshal but can't we just already only accept normalized paths in llb client parameters?

Reply:
The current implementation of the actions uses either a field called ```file``` ([```fileActionMkdir{}```](https://github.com/moby/buildkit/blob/80b91c55a47feb52a66e143a3929e614dd5c7adc/client/llb/fileop.go#L162), [```fileActionMkfile{}```](https://github.com/moby/buildkit/blob/80b91c55a47feb52a66e143a3929e614dd5c7adc/client/llb/fileop.go#L336), [```fileActionRm()```](https://github.com/moby/buildkit/blob/80b91c55a47feb52a66e143a3929e614dd5c7adc/client/llb/fileop.go#L404)) or ```dest``` ([```fileActionCopy{}```](https://github.com/moby/buildkit/blob/80b91c55a47feb52a66e143a3929e614dd5c7adc/client/llb/fileop.go#L498)). We would need to expose those fields as something that needs to be "normalized" before we call ```toProtoAction()```  here:

https://github.com/moby/buildkit/blob/80b91c55a47feb52a66e143a3929e614dd5c7adc/client/llb/fileop.go#L752

I can change this up whichever way you prefer, but the change would not be trivial, and may be better suited for a different PR.

Reply:
I renamed ```platform``` to ```platformOS``` (to avoid clashes with package names). Let me know if you prefer something else.